dataactbroker/handlers/fileHandler.py

```diff
@@ -859,31 +871,39 @@ def getProtectedFiles(self):
         response["urls"] = self.s3manager.getFileUrls(bucket_name=CONFIG_BROKER["static_files_bucket"], path=CONFIG_BROKER["help_files_path"])
         return JsonResponse.create(StatusCode.OK, response)
 
-    def completeGeneration(self, generationId):
+    def completeGeneration(self, generationId, file_type=None):
         """ For files D1 and D2, the API uses this route as a callback to load the generated file.
         Requires an 'href' key in the request that specifies the URL of the file to be downloaded
 
         Args:
             generationId - Unique key stored in file_generation_task table, used in callback to identify which submission
             this file is for.
+            file_type - the type of file to be generated, D1 or D2. Only used when calling completeGeneration for local development
 
         """
         try:
             if generationId is None:
                 raise ResponseException(
                     "Must include a generation ID", StatusCode.CLIENT_ERROR)
 
-            # Pull url from request
-            request_dict = RequestDictionary.derive(self.request)
-            _smx_logger.debug('Request content => %s', request_dict)
+            if not self.isLocal:
+                # Pull url from request
+                request_dict = RequestDictionary.derive(self.request)
+                _smx_logger.debug('Request content => %s', request_dict)
 
-            if 'href' not in request_dict:
-                raise ResponseException(
-                    "Request must include href key with URL of D file",
-                    StatusCode.CLIENT_ERROR
-                )
-            url = request_dict['href']
-            _smx_logger.debug('Download URL => %s', url)
+                if 'href' not in request_dict:
+                    raise ResponseException(
+                        "Request must include href key with URL of D file",
+                        StatusCode.CLIENT_ERROR
+                    )
+
+                url = request_dict['href']
+                _smx_logger.debug('Download URL => %s', url)
+            else:
+                if file_type == "D1":
+                    url = CONFIG_SERVICES["d1_file_path"]
+                else:
+                    url = CONFIG_SERVICES["d2_file_path"]
```

Comment:
They're local files so the file path will be unique to the person every time. Right now, for me, they're in the same place as the test files for integration tests but other people may have other preferences, plus they can name them what they want. I think we would run into more problems than it's worth to load the files from the same place every time.